src/main/java/org/myrobotlab/math/MapperLinear.java

```diff
@@ -0,0 +1,237 @@
+    this.maxOutput = maxY;
+    double t = minY;
+    if (invert) {
+      if (minY < maxY) {
+        minY = maxY;
+        maxY = t;
+      }
+      this.minOutput = maxY;
+      this.maxOutput = minY;
+    } else {
+      if (minY > maxY) {
+        minY = maxY;
+        maxY = t;
+      }
+      this.minOutput = minY;
+      this.maxOutput = maxY;
+    }
+
+    inverted = invert;
+  }
+
+  /* (non-Javadoc)
+   * @see org.myrobotlab.math.MapperInterface#getMinOutput()
+   */
+  @Override
+  public double getMinOutput() {
+    return minOutput;
+  }
+
+  /* (non-Javadoc)
+   * @see org.myrobotlab.math.MapperInterface#getMaxOutput()
+   */
+  @Override
+  public double getMaxOutput() {
+    return maxOutput;
+  }
+
+  /* (non-Javadoc)
+   * @see org.myrobotlab.math.MapperInterface#setMinMaxOutput(double, double)
+   */
+  @Override
+  public void setMinMaxOutput(double minOutput, double maxOutput) {
+    this.minOutput = minOutput;
+    this.maxOutput = maxOutput;
+  }
+
+  /* (non-Javadoc)
+   * @see org.myrobotlab.math.MapperInterface#getMinInput()
+   */
+  @Override
+  public double getMinInput() {
+    return minInput;
+  }
+
+  /* (non-Javadoc)
+   * @see org.myrobotlab.math.MapperInterface#getMaxInput()
+   */
+  @Override
+  public double getMaxInput() {
+    return maxInput;
+  }
+
+  /* (non-Javadoc)
+   * @see org.myrobotlab.math.MapperInterface#setMinMaxInput(double, double)
+   */
+  @Override
+  public void setMinMaxInput(double minInput, double maxInput) {
+    this.minInput = minInput;
+    this.maxInput = maxInput;
+  }
+
+  public String toString() {
+    return String.format(" < i < %.2f => map(%.2f,%.2f,%.2f,%.2f) => %.2f < o < %.2f ", minInput, maxInput, minX, maxX, minY, maxY, minOutput, maxOutput);
+  }
+
+  public void map(double minX, double maxX, double minY, double maxY) {
+    if (maxY - minY == 0) {
+      log.error("maxY - minY == 0 ..  DIVIDE BY ZERO ERROR COMING !");
+    }
+
+    this.minX = minX;
+    this.maxX = maxX;
+    this.minY = minY;
+    this.maxY = maxY;
+
+    if (minY < maxY) {
+      setInverted(false);
+    } else {
+      setInverted(true);
+    }
+  }
+  
+  /* (non-Javadoc)
+   * @see org.myrobotlab.math.MapperInterface#merge(org.myrobotlab.math.MapperLinear)
+   */
+  @Override
+  public void merge(MapperInterface in) {
+    if (MapperLinear.class.isAssignableFrom(in.getClass())) {
+      MapperLinear other = (MapperLinear)in; 
+    if (minX == null) minX = other.minX;
+    if (maxX == null) maxX = other.maxX;
+    if (minY == null) minY = other.minY;
+    if (maxY == null) maxY = other.maxY;
+    if (minInput == null) minInput = other.minInput;
+    if (maxInput == null) maxInput = other.maxInput;
+    if (minOutput == null) minOutput = other.minOutput;
+    if (maxOutput == null) maxOutput = other.maxOutput;
+    } else {
+      log.error("don't know how to merge %s into a MapperLinear mapper", in.getClass().getCanonicalName());
+    }
+  }
+
+  public static void main(String[] args) {

```

Comment:
move this to a unit test :)

Reply:
Done ...  and much more put into the unit test MapperLinearTest.java